src/odemis/driver/scanner.py

```diff
@@ -101,10 +101,12 @@ def __init__(self, name, role, children=None, dependencies=None, daemon=None, **
         self._va_external_ctrl = None
         self._va_blanker_ctrl = None
 
-        if "detector" in children:
-            ckwargs = children["detector"]
-            self._detector = CompositedDetector(parent=self, daemon=daemon, **ckwargs)
-            self.children.value.add(self._detector)
+        for cname, ckwargs in children.items():
+            if not cname.startswith("detector"):
+                raise ValueError(f"Only children detector* are supported, but got '{cname}'")
```

Comment:
```suggestion
                raise ValueError(f"Only children of type detector are supported, but got '{cname}'")
```
What exactly is the purpose of the * in this str?

Reply:
It's to mean that the role doesn't have to be "detector" exactly. Anything that starts with "detector" is fine.
I'll try to phrase that in English... without being too verbose.